qemu/tests/virtio_console.py

```diff
@@ -1632,23 +1646,49 @@ def test_rw_notconnect_guest():
         guest_worker = qemu_virtio_port.GuestWorker(vm)
         virtio_test.cleanup(vm, guest_worker)
 
+    @error_context.context_aware
     def test_rmmod():
         """
         Remove and load virtio_console kernel module.
         :param cfg: virtio_console_params - which type of virtio port to test
-        :param cfg: virtio_port_spread - how many devices per virt pci (0=all)
         """
-        (vm, guest_worker, port) = virtio_test.get_vm_with_single_port(
-            params.get('virtio_console_params'))
-        guest_worker.cleanup()
+        if params.get('virtio_console_params') == 'serialport':
+            vm = virtio_test.get_vm_with_ports(no_serialports=2)
+            _ports, ports = virtio_test.get_virtio_ports(vm)
+            guest_worker = qemu_virtio_port.GuestWorker(vm)
+        else:
+            (vm, guest_worker, ports) = virtio_test.get_vm_with_single_port(
+                params.get('virtio_console_params'))
+            guest_worker.cleanup()
+
         session = vm.wait_for_login()
-        if session.cmd_status('lsmod | grep virtio_console'):
+        if not session.cmd_output('lsmod | grep virtio_console'):
```

Comment:
Put these cmd into cfg, maybe?
Let's change later after other changes, if you consider put them into cfg